### ISSUE TITLE
Keywords trip up the parser in unexpected ways

### ISSUE THREAD

**Issue by lorefnon** [DanielXMoore/Civet#79]
There are many cases where keywords cause the parsing to fail: 

```
S.obj nullable: false
```

fails to parse because "able" is unexpected after `null`

Also tangentially, quoting `nullable` doesn't help

Similarly `S.obj true: false` fails to parse. 

If I update the NullLiteral to be defined as : 

```
NullLiteral
  /\b(null)\b/ ->
    return { $loc, token: $1 }
```

The nullable issue goes away, but things break if I add wordbreak boundaries to other literals like true/false. Perhaps there is a better/recommended way to handle this in hera ?

**Comment by lorefnon** [comment]
Not necessarily recommending we use them, but conventional tokenizers like [moo](https://github.com/no-context/moo) handle this well enough that you don't have to think about these things when defining parsers at all.

**Comment by edemaine** [comment]
This should be easy to fix with some negative assertions (same as what moo does it seems) or `\b`s. I think I've seen a few places in the grammar where I've wondered whether this is an issue; thanks for specific bug reports!

**Comment by edemaine** [comment]
Actually, looking at `true` and `false` shows the intended solution here: append the negative lookahead assertion `NonIdContinue`.

**Comment by STRd6** [comment]
Thanks for the bug report, fixed in 0.5.8!

**Comment by lorefnon** [comment]
Thank you
